### ISSUE TITLE
Improve reporting from guess

### ISSUE THREAD

**Issue by taldcroft** [taldcroft/asciitable#9]
The default guess functionality catches all exceptions and in the case of a failed table read just reports one exception at the end.  Add a mechanism to store the exception message for each attempted format and display this in a user friendly manner.  This would also help in cases such as issue #8 where the underlying problem of bad values for the converters input was hidden within guess.

**Comment by hamogu** [comment]
In one of my scripts I actually had a backwards compatibility problem with asciitable, which was caused by this behaviour of `guess`. 
I read tables which are produced by an external program (NX2 Racing by Silva Marine), which had faulty headers in earlier versions. Without `guess` I can read the table and if it throws an `InconsistentTableError` it is old format, if not, it's new format. With `guess` it just turned it into a one-column table.
That is easy to fix, of course (I just used `guess=False`), I only wanted to report that I also had trouble with exceptions which were hidden by `guess`.

**Comment by taldcroft** [comment]
I've pushed a possible implementation of the improved reporting from guess to the master branch.  Please give a try and see if it is helpful.

```
import asciitable
a = asciitable.read('1\n1 2')
a = asciitable.read('1\n1 2', delimiter=' ')
```

**Comment by taldcroft** [comment]
Included in 0.7.1.
